### PR TITLE
Story complete

### PR DESCRIPTION
**Pull request by vadam85**
Finished user story.
Added extension methods for checking user input.

### CODE REVIEW

File: rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    [Collection("TestCollection")]
+    public class AdminControllerTest
+    {
+        private readonly TestContext _testContext;
+
+        public AdminControllerTest(TestContext testContext)
+        {
+            this._testContext = testContext;
+        }
+        [Fact]
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs:23]: styleguide: put an empty line before [Fact] please

File: rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        public async Task Add_Should_Return_Created()
+        {
+            var request = new AddAdminPostRequestMock(new AddAdminPostRequestMockBody().SetCorrectBody()).SetCorrectHeaders();
+            var response = await _testContext.Client.PostAsync("/admin/add", request);
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+            Assert.Equal(JsonConvert.SerializeObject(new ResponseMessage("Success")), await response.Content.ReadAsStringAsync());
+        }
+        [Fact]
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs:32]: styleguide: put an empty line before [Fact] please

File: rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        public async Task Incorrect_ContentType_Should_Return_UnsupportedMediaType()
+        {
+            var request = new AddAdminPostRequestMock(new AddAdminPostRequestMockBody().SetCorrectBody()).SetMissingContentType();
+            var response = await _testContext.Client.PostAsync("/admin/add", request);
+            Assert.Equal(HttpStatusCode.UnsupportedMediaType, response.StatusCode);
+
+            response = await _testContext.Client.PostAsync("/admin/add", request.SetXmlContentType());
+            Assert.Equal(HttpStatusCode.UnsupportedMediaType, response.StatusCode);
+        }
+        [Fact]
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs:43]: styleguide: put an empty line before [Fact] please

File: rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            var request = new AddAdminPostRequestMock(new AddAdminPostRequestMockBody().SetCorrectBody()).SetMissingUsertokenauth();
+            var response = await _testContext.Client.PostAsync("/admin/add", request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+            Assert.Equal(JsonConvert.SerializeObject(new ErrorMessage("usertokenauth missing")), await response.Content.ReadAsStringAsync());
+
+            response = await _testContext.Client.PostAsync("/admin/add", request.SetEmptyUsertokenauth());
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+            Assert.Equal(JsonConvert.SerializeObject(new ErrorMessage("usertokenauth missing")), await response.Content.ReadAsStringAsync());
+        }
+        [Fact]
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs:56]: styleguide: put an empty line before [Fact] please

File: rueppellii-lvlup-asp.net-core/Controllers/AdminController.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        [HttpPost("add")]
+        [Consumes("application/json")]
+        public IActionResult Add(AddAdminDto addAdminDto)
+        {
+            if (string.IsNullOrEmpty(Request.Headers["usertokenauth"]))
+            {
+                return StatusCode(401, new ErrorMessage("usertokenauth missing"));
+            }
+            if(addAdminDto.IsAnyPropertyNull() || addAdminDto.IsAnyStringPropertyEmpty())
```
Comment (szirmi) [inline rueppellii-lvlup-asp.net-core/Controllers/AdminController.cs:20]: i can't see the point in this pull request where these 2 methods connecting to AddAdminDto class
don't you need to use UserInputChecker.IsAnyPropertyNull(addAdminDto) instead?

**Review by szirmi — CHANGES_REQUESTED** [review]

File: rueppellii-lvlup-asp.net-core/Controllers/AdminController.cs (hunk 5 above)
Comment (vadam85) [reply rueppellii-lvlup-asp.net-core/Controllers/AdminController.cs:20]: These two methods are described in the Extensions folder / namespace.
These are extension methods on the Object class. Süvi mentioned we should do something like this.

Comment (szirmi) [reply rueppellii-lvlup-asp.net-core/Controllers/AdminController.cs:20]: okay then 👍 
thanks for the information

File: rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs (hunk 1 above)
Comment (vadam85) [reply rueppellii-lvlup-asp.net-core.IntegrationTests/Scenarios/AdminControllerTest.cs:23]: Fixed the empty lines / styleguide in the Test.

**Review by szirmi — APPROVED** [review]

**Pull request merged**